### DISCUSSION TITLE
Why is my smart query returning every single object ?

### DISCUSSION THREAD

**Discussion by BrannJoly** [mikro-orm/mikro-orm#4141] [Q&A] [answered]
I have an Entity define as such : 

```
@Entity({ tableName: 'project' })
export class ProjectDbEntity {
  @PrimaryKey()
  id: ProjectId = uuidv4();

  @Unique()
  @Property()
  slug: string;

 // many other fields ommited
}

```

Upon insertion of a new project, I want to make sure the slug is unique, so I want to get all the other projects that start with the generated slug in order to append a number at the end of my slug if needed to ensure uniqueness

This is the smart query I'm using : 
`    let startWithSameSlugProjects = await this.projectEntityRepository.findAll({ slug: { $like: projectEntity.slug } } as any);
`

This returns me every single project in my database. Also, I add to cast as any to get this to transpile. Otherwise, typescript was complaining that : 
` Argument of type '{ slug: { $like: string; }; }' is not assignable to parameter of type 'FindOptions<ProjectDbEntity, never>'.
  Object literal may only specify known properties, and 'slug' does not exist in type 'FindOptions<ProjectDbEntity, never>'.` 

I'm using PostgresSql and mikroOrm 8.7.7

Is there something I'm doing wrong ?
Thanks in advance !

**Comment by B4nan** [comment, marked as answer]
Don't use `findAll`, that method indeed finds all the rows, there is no filtering support - while you are trying to put the query there, instead of the expected `FindOptions` object.

```diff
-let startWithSameSlugProjects = await this.projectEntityRepository.findAll({ slug: { $like: projectEntity.slug } } as any);
+let startWithSameSlugProjects = await this.projectEntityRepository.find({ slug: { $like: projectEntity.slug } });
```

**Reply by BrannJoly** [reply]
> Don't use `findAll`, that method indeed finds all the rows, there is no filtering support - while you are trying to put the query there, instead of the expected `FindOptions` object.

... and now I feel stupid :) thank you for your help ! (and for a great ORM)
